neural_compressor/adaptor/torch_utils/layer_wise_quant/modified_pickle.py

```diff
@@ -0,0 +1,1802 @@
+"""Create portable serialized representations of Python objects.
+
+See module copyreg for a mechanism for registering custom picklers.
+See module pickletools source for extensive comments.
+
+Classes:
+
```

Comment:
please be careful of the license. I assume most of the code is borrowed from other lib.